### PR TITLE
IS-203 OIDC claims and scopes

### LINKED ISSUE #203: OIDC: Add claims and scopes specification for Sweden Connect
Building on OIDC Sweden, we need to add additional claims and scopes for Sweden Connect.

### PR DESCRIPTION
**Pull request by martin-lindstrom**
First draft of OIDC claims and scopes specification focusing on eIDAS attributes.

Closes #203

### CODE REVIEW

File: OIDC - Sweden Connect - Claims and Scopes Specification.md [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+**Scope:** `https://id.swedenconnect.se/scope/eidasNaturalPersonIdentity`
+
+**Description:** A scope that defines a claim set that provides identity information for a natural person authenticated via the Swedish eIDAS-node (via a eIDAS member state). 
+
+| Requested Claims | Description/comment | Reference |
+| :--- | :--- | :--- |
+| `https://id.swedenconnect.se/claim/prid` | Provisional identifier | This specification - \[[2.1.1](#provisional-identifier)\] |
+| `https://id.swedenconnect.se/claim/pridPersistence` | Persistence indicator for the above `prid` claim | This specification - \[[2.1.2](#provisional-identifier-persistence-indicator)\] |
+| `https://id.swedenconnect.se/claim/eidasPersonIdentifier` | eIDAS Person Identifier | This specification - \[[2.1.6](#eidas-person-identifier)\] |
+
```
Comment (Razumain) [inline OIDC - Sweden Connect - Claims and Scopes Specification.md:199]: add https://id.swedenconnect.se/claim/eidasPersonIdentifier

File: OIDC - Sweden Connect - Claims and Scopes Specification.md [hunk 2]
```diff
@@ -0,0 +1,363 @@
...
+<a name="additional-eidas-claims"></a>
+#### 3.1.3. Additional eIDAS Claims
+
+Not all eIDAS attributes/claims listed in [Appendix A](#conversion-of-eidas-attributes) are covered by the above scope definitions. This section is an informational text informing Relying Parties how to obtain additional claims for an eIDAS authentication.
+
+- To obtain name and date of birth claims, include the `https://id.oidc.se/scope/naturalPersonInfo` scope (section 3.1 of  \[[OIDC.Sweden.Claims](#oidc-sweden-claims)\]).
+
+- For the eIDAS Person Identifier, which is the eIDAS identity delivered from the foreign member state, include an explicit claim request for the `https://id.swedenconnect.se/claim/eidasPersonIdentifier` claim ([2.1.6](#eidas-person-identifier).
+
```
Comment (Razumain) [inline OIDC - Sweden Connect - Claims and Scopes Specification.md:255]: Include this in the https://id.swedenconnect.se/scope/eidasNaturalPersonIdentity scope

File: OIDC - Sweden Connect - Claims and Scopes Specification.md [hunk 3]
```diff
@@ -0,0 +1,363 @@
...
+**Note:** None of the claims are marked as "essential" since the eIDAS Connector will only deliver the claims if an identity binding exists. Also, the scope definition states that the claims should be delivered via the UserInfo endpoint and not directly in the ID Token. The reason for this is that a mapped identity can never be seen as a primary eIDAS identity (since the claims are only delivered if a binding exists).
+
+**Note (ii):** The `mappedPersonalIdentityNumber` and `mappedCoordinationNumber` claims are mutually exclusive. A user has bound his or hers eIDAS identity to a Swedish civic registration number ("personnummer") **or** a Swedish coordination number ("samordningsnummer"), never both.
+
+<a name="additional-eidas-claims"></a>
+#### 3.1.3. Additional eIDAS Claims
+
+Not all eIDAS attributes/claims listed in [Appendix A](#conversion-of-eidas-attributes) are covered by the above scope definitions. This section is an informational text informing Relying Parties how to obtain additional claims for an eIDAS authentication.
+
```
Comment (Razumain) [inline OIDC - Sweden Connect - Claims and Scopes Specification.md:251]: We have an obligation according to the eIDAS directive to provide a minimum data set. This includes the eIDASPersonIdentifier, given name, surname and date of birth.

We should have a text explaining what scopes you need to include in order to get this identity information.
Perhaps, we should also say that the eIDAS connector MAY choose to reject a request if the minimum data set is not requested (We have to check this, but the eIDAS connector may be obligated to deliver this).

File: OIDC - Sweden Connect - Claims and Scopes Specification.md [hunk 4]
```diff
@@ -0,0 +1,363 @@
...
+  },
+  "id_token" : {
+    "https://id.swedenconnect.se/claim/prid" : { "essential" : true },
+    "https://id.swedenconnect.se/claim/pridPersistence" : { "essential" : true }
+  }
+}
+```
+
+**Note:** It is RECOMMENDED that Swedish Relying Parties use the `https://id.swedenconnect.se/claim/prid` claim as a primary identity for a user identified using eIDAS. Should a Relying Party should want to use the eIDAS Person Identifier instead, it should include an explicit claim request for the `https://id.swedenconnect.se/claim/eidasPersonIdentifier` claim in its authentication request.
+
```
Comment (Razumain) [inline OIDC - Sweden Connect - Claims and Scopes Specification.md:216]: See above

**Review by Razumain — CHANGES_REQUESTED** [review]

File: OIDC - Sweden Connect - Claims and Scopes Specification.md (hunk 3 above)
Comment (martin-lindstrom) [reply OIDC - Sweden Connect - Claims and Scopes Specification.md:251]: This specification is a claims and scope spec. If we have any specific requirements concerning the use of eIDAS we will have to add that in the main profile.

**Pull request merged**